httpql/src/main/java/com/hubspot/httpql/ann/FilterBy.java

```diff
@@ -25,4 +25,6 @@
   Class<? extends Filter>[] value();
 
   String as() default "";
+
+  Class<?> typeOverride() default void.class;
```

Comment:
I'm actually not sure I understand what this annotation param does. Could you add it to the docstring above and to the tests?